test/13_arraycommon.jl

```diff
@@ -979,4 +979,121 @@ end
     @test x == [3, 3, 3]
 end
 
+@testset "append! ordered=$ordered reftype=$R" for ordered in (false, true),
+    R in (CategoricalArrays.DefaultRefType, UInt8, UInt, Int8, Int)
```

Comment:
But does it hurt? I would retain it.

Reply:
It doesn't really hurt but it makes the run take longer, and there's really no reason why it would work for one `R` type and not another. If you really want to test corner cases with `R`, I'd rather check that `append!` works when the inputs have different type parameters -- but even that sounds obvious given how the function is implemented.

Reply:
OK - I have removed `R` from tests.